### DISCUSSION TITLE
Printing using expo print Api

### DISCUSSION THREAD

**Discussion by LouisMuriuki** [expo/expo#19481] [Archived: Questions] [answered]
I am building a POS app and I want to be able to print a receipt, I have gone through the expo print API documentation  and even implemented its functionality in my app, as of now I can export a html file into PDF which is good but I want to be able to select a printer and print that pdf .However I don't not have a printer with me yet. Here  is my query, so when I get a Bluetooth printer and connect it to my phone will the API be able to see the printer and actually print the specific file?

**Comment by amilaupendra** [comment]
did you find an answer? how to do that?

**Reply by LouisMuriuki** [reply, marked as answer]
Hello @amilaupendra 
yes i did, However expo print wont work.
I used this library https://www.npmjs.com/package/react-native-thermal-printer .
note. if you are using managed workflow you will have to prebuild your app then you can use the package
you can read here :
1 https://docs.expo.dev/workflow/customizing/
2. https://docs.expo.dev/workflow/prebuild/
hope this helps

**Reply by Rokhel121314** [reply]
@LouisMuriuki I'm also trying to make a pos system and create a print receipt feature and having trouble with the html variable. Is it possible to print a react native component where the codes for receipt located. did you convert the component into html how did you do it?

**Reply by LouisMuriuki** [reply]
Hello @Rokhel121314 are you using this library https://www.npmjs.com/package/react-native-thermal-printer ?

**Reply by eltonvilanculo** [reply]
I've been working on this issue for almost 2 weeks and definitly you should use https://www.npmjs.com/package/react-native-thermal-printer

**Reply by amilaupendra** [reply]
Update: i successfully implemented an android mobile app using [https://github.com/januslo/react-native-bluetooth-escpos-printer](url) package. also i made a video on youtube how to do it. if anyone interested please watch [https://youtu.be/BXbseqogkMQ?si=BeVEKeNMlN3a3Sj2](url). you can make it using expo or react native cli.  testted it on android version 5.1 to 13.

**Comment by ghost96-prog** [comment]
hie, um getting this error

Cannot read property 'printBluetooth' of null
at node_modules\react-native-thermal-printer\src\index.tsx:null in printBluetooth
at app\screens\Settings.js:null in onPress
at node_modules\@babel\runtime\helpers\asyncToGenerator.js:null in asyncGeneratorStep
at node_modules\@babel\runtime\helpers\asyncToGenerator.js:null in _throw
at node_modules\@babel\runtime\helpers\asyncToGenerator.js:null in _asyncToGenerator
at node_modules\@babel\runtime\helpers\asyncToGenerator.js:null in _next
at app\screens\Settings.js:null in Settings
at node_modules\react-native\Libraries\Pressability\Pressability.js:null in _performTransitionSideEffects
at node_modules\react-native\Libraries\Pressability\Pressability.js:null in _receiveSignal
- ... 21 more stack frames from framework internals
-

**Comment by AYMANEMTS** [comment]
I have problem in style recipet ticket is take many spaces top
